plutus-wallet-api/src/Language/PlutusTx/StateMachine.hs

```diff
@@ -66,3 +71,14 @@ mkValidator (StateMachine step check) currentState (input, unseal -> HashedDataS
             traceIfFalseH "State transition invalid - checks failed"
             (check currentState input ptx)
     in dataScriptOk && stateOk && checkOk
+
+-- | Given a state machine, take a data script wrapping the old state and an
+-- input and compute the data script wrapping the new state.
+mkData :: Lift i => StateMachine s i -> DataScript -> i -> DataScript
+mkData sm (DataScript script) input = DataScript $
+    $$(Scripts.compileScript [|| \(s :: s) (i :: i) ->
+        case smTransition sm s i of
```

Comment:
I thought we concluded this didn't work? And indeed this function doesn't look like it can work, it has a free variable!

Reply:
I thought I reverted that commit.... hmmm... let me check!

Reply:
I don't know where you found that snippet, @michaelpj , put it's not included in this PR.

Reply:
Huh, it's the first commit, but I guess you reverted it and then turned the revert into the main second commit? Confusing. (FWIW I normally just rebase out commits if they haven't gone into master yet, since reverts are often confusing!)

Reply:
Yeah, sorry about that. I thought I had sqashed everything into one commit.